src/main.rs

```diff
@@ -6,7 +6,13 @@ use toipe::Toipe;
 fn main() {
     let config = ToipeConfig::parse();
 
-    let mut toipe = Toipe::new(config).unwrap();
+    let res = Toipe::new(config);
+    if let Err(e) = res {
+        eprintln!("{}", e);
+        std::process::exit(1);
+    }
+
+    let mut toipe = res.unwrap();
```

Comment:
This is not very nice, perhaps not "idiomatic" rust.

Instead, you can change the main function's signature to be:
```rust
fn main() -> Result<(), ToipeError>
```

This will allow us to use the `?` operator to stop at an error and print it. See the last code block in [this section](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#where-the--operator-can-be-used) of the rust book.

Though, I haven't tried this out, see if it works.

Reply:
it does work, so i changed it to they way you wanted.